pilot/pkg/networking/core/v1alpha3/extension_config_builder.go

```diff
@@ -29,7 +30,10 @@ func (configgen *ConfigGeneratorImpl) BuildExtensionConfiguration(
 ) []*core.TypedExtensionConfig {
 	envoyFilterPatches := push.EnvoyFilters(proxy)
 	extensions := envoyfilter.InsertedExtensionConfigurations(envoyFilterPatches, extensionConfigNames)
+
 	wasmPlugins := push.WasmPlugins(proxy)
 	extensions = append(extensions, extension.InsertedExtensionConfigurations(wasmPlugins, extensionConfigNames, pullSecrets)...)
+
+	extensions = append(extensions, telemetry.InsertedExtensionConfigurations(proxy, push, extensionConfigNames)...)
```

Comment:
we can add a filter to skip `telemetry.InsertedExtensionConfiguration` if not needed, which is inefficient

Reply:
it will skip if `extensionConfigNames` is empty

Reply:
NO, I mean we only need to call it when it is stats telemetry names.

Reply:
updated